platform/android/app/src/main/kotlin/coop/polypoly/polypod/network/Network.kt

```diff
@@ -54,16 +57,57 @@ class Network(val context: Context) {
             outputStream.flush()
         } catch (exception: Exception) {
             logger.error("network.httpPost failed: $exception")
-            return@withContext exception.toString()
+            response.error = exception.toString()
+            return@withContext response
+        }
+        val responseCode = connection.responseCode
+        if (responseCode < 200 || responseCode > 299) {
```

Comment:
As commented above, we should maybe be a bit more specific about status codes.

Reply:
This code has a bunch of duplication, I think it makes more sense to eliminate it later rather than hold this back further. But in terms of being specific - any HTTP response 200-299 is valid - anything else would be unexpected, I'd say the logic is solid (just duplicated).